Source/WebKit/UIProcess/API/gtk/PageClientImpl.cpp

```diff
@@ -632,4 +632,9 @@ WebCore::Color PageClientImpl::accentColor()
     return SRGBA<uint8_t> { 52, 132, 228 };
 }
 
+WebKitWebResourceLoadManager* PageClientImpl::webResourceLoadManager()
+{
+    return WEBKIT_IS_WEB_VIEW(m_viewWidget) ? webkitWebViewGetWebResourceLoadManager(WEBKIT_WEB_VIEW(m_viewWidget)) : nullptr;
```

Comment:
Can you remind me how we can get a `PageClientImpl` with a `m_viewWidget` that is not a `WebKitWebView`? It looks like it's only constructed from `WebKitWebViewBase`, and `PageClientImpl` does assume pervasively that it is of that type. Is it possible to have a `WebKitWebViewBase` that is not a `WebKitWebView`? Well, it is a public type, so I suppose an application could create one directly... but can it happen normally? If it's not normal, then I bet we can simplify things by removing more nullability.

Reply:
Yes, when using the C API you have a `WebKitWebViewBase` that is not a `WebKitWebView`, that's why we have base class. In the case of wpe the view client is not implemented for WTR.

Reply:
The web inspector also uses WebKitWebViewBase only